### ISSUE TITLE
SystemStackError (stack level too deep) with net_http hook

### ISSUE THREAD

**Issue by benjaminwols** [appsignal/appsignal-ruby#583]
## Describe the bug
Using appsignal in combination with http_logger results in an SystemStackError (stack level too deep)

## To Reproduce
- Using AppSignal for Ruby gem version 2.10.2
- In my app using gem http_logger version 0.6.0
- With this code:
  ```
  base_uri = URI.parse('https://www.google.com')
  req = Net::HTTP::Get.new base_uri
  Net::HTTP.start(base_uri.host, base_uri.port, use_ssl: true, read_timeout: 5) { |http| http.request(req) }
  ```

**Comment by benjaminwols** [comment]
Reported at http_logger initially and they suggest to use prepend instead of alias
https://github.com/railsware/http_logger/issues/18

**Comment by tombruijn** [comment]
@matsimitsu are you looking into this?

The problem with using `prepend` is that we drop 1.9 support as it's not available in Ruby 1.9. Discussion [here (private link)](https://github.com/appsignal/handbook/issues/155).

**Comment by benjaminwols** [comment]
@tombruijn @matsimitsu Is there an update on this issue?

**Comment by tombruijn** [comment]
Hi @benjaminwols, We haven't picked this up yet. We probably need to make an exception for Ruby 1.9 if we want to go the `prepend` approach for the monkeypatch.

**Comment by tombruijn** [comment]
Hi again @benjaminwols, I had a look at a similar issue today and I see the `Module.prepend` and aliasing method combination is causing more issues than just in Net::HTTP. I've created PR #597 that fixes your reported issue hopefully. Can you try it out (on a non-production environment) and let me know if it worked?

**Comment by tombruijn** [comment]
@benjaminwols heads up! We're going to revert the fix for this today in a new Ruby gem version, see PR #604. You can stay on the version that includes the fix for now while we figure out a better solution, if possible. You can follow the discussion about this issue in #603.

**Comment by tombruijn** [comment]
This fix has been reapplied in Ruby gem 3.0: https://blog.appsignal.com/2021/03/01/ruby-gem-3-0.html
